sormas-ui/src/main/java/de/symeda/sormas/ui/location/LocationEditForm.java

```diff
@@ -286,6 +286,35 @@ protected void addFields() {
 				facilityDetails.setRequired(false);
 				facilityDetails.clear();
 			}
+
+			// Fill in the address fields based on the selected facility
+			// We don't want the location form to automatically change even if the facility's address is updated later 
+			// on, so we only trigger it upon a manual change of the facility field
+			// We use isAttached() to avoid the fuss when initializing the form, it may seems a bit hacky, but it is
+			// necessary because isModified() will still return true for a short duration even if we keep the very same 
+			// value because of this field dependencies to other fields and the way updateEnumValues works
+			if (facility.isAttached()) {
+				if (facility.getValue() != null) {
+					FacilityDto facilityDto = FacadeProvider.getFacilityFacade().getByUuid(((FacilityReferenceDto) facility.getValue()).getUuid());
```

Comment:
I think blindly overwriting values here is not a good idea. Instead, I propose the following solution:

For all of these fields, check whether the facility has a value for it. If not, don't overwrite whatever has been entered in the location form. If yes, check whether the location form has a value for it. If no, overwrite that empty value with the value from the facility. If yes, check whether the value is identical; If yes, do nothing; If no, show a popup window that asks the user to confirm overwriting the location details with the details from the facility (or to reject; in that case, nothing is overwritten).

I suggest the following associated text:
"The selected facility has location details that are different from the location details you are currently editing. Do you want to overwrite the details of the current location with those from the facility?"